src/Core.UnitTests/Notifications/DisabledNotificationsStorageTests.cs

```diff
@@ -0,0 +1,136 @@
+                        
+            var result = testSubject.IsNotificationDisabled(notificationId);
+
+            result.Should().Be(expectedResult);
+            file.Verify(f => f.ReadAllLines(GetFilePath()), Times.Once);
+        }
+
+        [TestMethod]
+        public void DisableNotification_IsNotDisabled_Disables()
+        {
+            var file = CreateFileMock("1", "2", "3");
+            var testSubject = CreateTestSubject(file.Object);
+
+            var result = testSubject.IsNotificationDisabled("4"); 
+            result.Should().BeFalse();
+
+            testSubject.DisableNotification("4");
+
+            result = testSubject.IsNotificationDisabled("4");
+            
+            result.Should().BeTrue();
+            file.Verify(f => f.WriteAllLines(GetFilePath(), It.IsAny<IEnumerable<string>>()), Times.Once);
+            var disabledNotifications = (IEnumerable<string>)file.Invocations[1].Arguments[1];
+
+            disabledNotifications.Count().Should().Be(4);
+            disabledNotifications.Contains("4").Should().BeTrue();
+        }
+
+        [TestMethod]
+        public void DisableNotification_AlreadyDisabled_DoesNothing()
+        {
+            var file = CreateFileMock("1", "2", "3");
+            var testSubject = CreateTestSubject(file.Object);
+
+            var result = testSubject.IsNotificationDisabled("3");
+            result.Should().BeTrue();
+
+            testSubject.DisableNotification("3");
+
+            result = testSubject.IsNotificationDisabled("3");
+
+            result.Should().BeTrue();
+            file.Verify(f => f.WriteAllLines(It.IsAny<string>(), It.IsAny<IEnumerable<string>>()), Times.Never);
+        }
+
+        private IDisabledNotificationsStorage CreateTestSubject(IFile file)
+        {
+            var fileSystem = CreateFileSystem(file);
+            var versionProvider = CreateVsVersionProvider();
+
+            var testSubject = new DisabledNotificationsStorage(versionProvider, fileSystem);
+
+            return testSubject;
+        }
+
+        private IFileSystem CreateFileSystem(IFile file)
+        {
+            var fileSystem = new Mock<IFileSystem>();
+            fileSystem.SetupGet(fs => fs.File).Returns(file);
+
+            return fileSystem.Object;
+        }
+
+        private Mock<IFile> CreateFileMock(params string[] fileContent)
+        {
+            var file = new Mock<IFile>();
+            file.Setup(f => f.ReadAllLines(It.IsAny<string>())).Returns(fileContent);
+
+            return file;
+        }
+
+        private IVsVersionProvider CreateVsVersionProvider()
+        {
+            var version = new Mock<IVsVersion>();
+            version.SetupGet(v => v.MajorInstallationVersion).Returns("17");
+
+            var versionProvider = new Mock<IVsVersionProvider>();
+            versionProvider.SetupGet(vp => vp.Version).Returns(version.Object);
+
+            return versionProvider.Object;
+        }
+
+        private string GetFilePath()
```

Comment:
This shouldn't be copy-pasted, maybe it should be internal in the product code, or the final property should be internal.

Reply:
I did this way so if we have unintentionally changed the save path we would have a warning.

Reply:
A better name would be `GetExpectedDisabledNotificationsFilePath`.